### PR TITLE
xping 1.3

### PR DESCRIPTION
**Pull request by martintopholm**
The pull-request below is "self-submitted" work.

The following changes since commit 91b6b4c70365567b9df75df5b6806a9b49168d73:
 
  libcppa: update 0.13 bottle. (2015-03-12 12:41:04 +0000)

are available in the git repository at:

  https://github.com/martintopholm/homebrew.git xping

for you to fetch changes up to 2967aa7a52a17134cc790233b6313e94d6c0ef02:

  xping 1.3 (2015-03-12 14:35:08 +0100)

----------------------------------------------------------------
Martin Topholm (1):
      xping 1.3

 Library/Formula/xping.rb | 49 +++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 49 insertions(+)
 create mode 100644 Library/Formula/xping.rb

### CODE REVIEW

**Comment by mikemcquaid** [comment]
Probably worth maintaining this in your own tap instead but I'll make a few comments anyway.

File: Library/Formula/xping.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+require "formula"
```
Comment (mikemcquaid) [inline Library/Formula/xping.rb]: Don't need this any more; run `brew audit --strict`

File: Library/Formula/xping.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+class Xping < Formula
+
```
Comment (mikemcquaid) [inline Library/Formula/xping.rb]: Remove this empty line.

File: Library/Formula/xping.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  homepage "http://github.com/martintopholm/xping"
+  version "1.3"
+  depends_on "libevent"
+
+  stable do
+    url "https://github.com/martintopholm/xping/archive/v#{version}.tar.gz"
+    sha1 "822cd76851af577d2b987dd65933ca9187a4ea8f"
+    patch do
+      url "http://martin.topholm.eu/pub/xping/brew/0001-main-keep-env-cflags-and-clear-install-deps.patch"
```
Comment (mikemcquaid) [inline Library/Formula/xping.rb]: This patch should be committed to the upstream project.

File: Library/Formula/xping.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
+require "formula"
+
+class Xping < Formula
+
+  homepage "http://github.com/martintopholm/xping"
+  version "1.3"
+  depends_on "libevent"
```
Comment (mikemcquaid) [inline Library/Formula/xping.rb]: Newline before this line.

File: Library/Formula/xping.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+  stable do
+    url "https://github.com/martintopholm/xping/archive/v#{version}.tar.gz"
+    sha1 "822cd76851af577d2b987dd65933ca9187a4ea8f"
+    patch do
```
Comment (mikemcquaid) [inline Library/Formula/xping.rb]: newline before this line.

File: Library/Formula/xping.rb [hunk 6]
```diff
@@ -0,0 +1,49 @@
+require "formula"
+
+class Xping < Formula
+
+  homepage "http://github.com/martintopholm/xping"
+  version "1.3"
+  depends_on "libevent"
+
+  stable do
+    url "https://github.com/martintopholm/xping/archive/v#{version}.tar.gz"
```
Comment (mikemcquaid) [inline Library/Formula/xping.rb]: hardcode the `url`; don't use `version`

File: Library/Formula/xping.rb [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    sha1 "822cd76851af577d2b987dd65933ca9187a4ea8f"
```
Comment (mikemcquaid) [inline Library/Formula/xping.rb]: Use `sha256`

File: Library/Formula/xping.rb [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    patch do
+      url "http://martin.topholm.eu/pub/xping/brew/0001-main-keep-env-cflags-and-clear-install-deps.patch"
+      sha1 "c52dd1d313a0b0f8055f0669effbddf14cae067e"
+    end
+    patch do
```
Comment (mikemcquaid) [inline Library/Formula/xping.rb]: newline before this line.

File: Library/Formula/xping.rb [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+      url "http://martin.topholm.eu/pub/xping/brew/0003-termio-correct-labelwidth-variable-scope.patch"
+      sha1 "dfac12bbf127aad28ed9702c5d08ecd3996256a6"
+    end
+  end
+  head do
```
Comment (mikemcquaid) [inline Library/Formula/xping.rb]: newline before this line

File: Library/Formula/xping.rb [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    url "https://github.com/martintopholm/xping.git", :using => :git, :branch => "master"
+  end
+
+  def install
+    ENV.append "DEPS", "check-curses.c"
+    ENV.append "CFLAGS", "-DNCURSES"
+    ENV.append "LIBS", "-lncurses"
```
Comment (mikemcquaid) [inline Library/Formula/xping.rb]: Why do these all need to be set?

File: Library/Formula/xping.rb [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+    system "make"
+    system "make install PREFIX=#{prefix} MANPATH=#{prefix}/share/man"
```
Comment (mikemcquaid) [inline Library/Formula/xping.rb]: Split system args `"make", "install", "...`

File: Library/Formula/xping.rb [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+  end
+
+  test do
+    system "#{bin}/xping -V"
```
Comment (mikemcquaid) [inline Library/Formula/xping.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/xping.rb [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+  end
+
+  def caveats; <<-EOS.undent
+    xping requires superuser privileges. You can either run the program
+    via `sudo`, or change its ownership to root and set the setuid bit:
```
Comment (mikemcquaid) [inline Library/Formula/xping.rb]: We don't recommend `setuid` ever for security reasons.

**Comment by martintopholm** [comment]
Hi Mike, thank you for your feedback.

The tap is available as

    brew tap martintopholm/xping

**Pull request closed without merging**